tests/libcxxthrd/enc/enc.cpp

```diff
@@ -114,30 +127,115 @@ static int _pthread_join_hook(pthread_t enc_thread, void** retval)
         });
     if (it == _key_to_thread_id_map.end())
     {
-        printf("Enclave Key for thread ID 0x%lu not found\n", enc_thread);
+        printf(
+            "_pthread_join_hook(): Error: enc_key for thread ID 0x%lu not "
+            "found\n",
+            enc_thread);
         oe_abort();
     }
-    join_enc_key = it->first;
+    int join_enc_key = it->first;
+    ThreadArgs* thrd_join_args = thread_args[join_enc_key - 1];
+    if (thrd_join_args == NULL)
+    {
+        _release_lock(&_enc_lock);
+        return EINVAL;
+    }
+    thrd_join_args->enc_key = join_enc_key;
+    thrd_join_args->join_value_ptr = value_ptr;
     _release_lock(&_enc_lock);
 
     printf(
-        "_pthread_join_hook(): Enclave Key for thread ID 0x%lu is %d\n",
+        "_pthread_join_hook(): enc_key for thread ID 0x%lu is %d\n",
         enc_thread,
-        join_enc_key);
-    if (oe_call_host("host_join_pthread", (void*)(uint64_t)join_enc_key) !=
-        OE_OK)
+        it->first);
+    if (oe_call_host("host_join_pthread", (void*)thrd_join_args) != OE_OK)
         oe_abort();
 
-    return 0;
+    // pthread_join is blocking. So, wait until host returns a value
+    int join_ret = -1;
+    while (join_ret == -1)
```

Comment:
oe_call_host is a blocking call. So why do you need this loop here? Control will reach 156 in the current thread only after the host returns in line 151.

Reply:
Fixed the printf statement to indicate this